pytorch_lightning/plugins/environments/slurm_environment.py

```diff
@@ -28,6 +28,11 @@ class SLURMEnvironment(ClusterEnvironment):
     def creates_processes_externally(self) -> bool:
         return True
 
+    @staticmethod
+    def is_using_slurm() -> bool:
```

Comment:
do we want this to be public? do you see users needing to access the property?

as a SLURM noob, this is a bit confusing, because it's in the `SLURMEnvironment` class so I would expect to always return `True`.

Is there a better name.

Reply:
what about the `detect` method from https://github.com/PyTorchLightning/pytorch-lightning/issues/6303 ? 
```
@abstractmethod
@staticmethod
def detect() -> bool:
```

Reply:
@carmocca 
The current choice is simply following the existing pattern we have for torchelastic and others.

I agree that we want a detect() method. 
We can name it detect() here and update the others in a new PR. But we can't add it to the base and make it abstract in this PR.

Reply:
#10564